lighthouse-core/gather/driver.js

```diff
@@ -24,6 +25,21 @@ const pageFunctions = require('../lib/page-functions.js');
 // eslint-disable-next-line no-unused-vars
 const Connection = require('./connections/connection.js');
 
+const UIStrings = {
+  /**
+   * @description Warning that important data was not cleared but may have affected the scores of this run.
+   * @example {IndexedDB, Local Storage} locations
+   */
+  warningData: `{locationCount, plural,
+    =1 {There may be important data in this location: {locations}. ` +
```

Comment:
maybe some variation on
`There may be stored data affecting loading performance in this location: {locations}. Audit this page in an incognito window to prevent those resources from affecting your scores.`
?
(and similarly for the plural string)

It seems important to call out the effect on performance (why we're warning at all) right at the start